src/swish/osi.c

```diff
@@ -239,6 +233,19 @@ static ptr close_fs_port(uptr port, ptr callback) {
   return Strue;
 }
 
+static ptr close_fd_port(uptr port, ptr callback) {
+  fs_port_t* p = (fs_port_t*)port;
+  free(p);
+  osi_add_callback1(callback, Sfixnum(0));
```

Comment:
Clever how our `.close` vtable entries avoid unnecessary lock/unlock of the `callback`. I hadn't noticed that before doing a double-take here.

Reply:
Yes, because the callback list is held in Scheme's heap, we don't need the lock!